packages/components/components/elvis-segmented-control/src/react/useSegmentedControlHighlighter.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { SegmentedControlProps } from './elviaSegmentedControl.types';
+
+export const useSegmentedControlHighlighter = (
+  selectedIndex: number,
+  items: SegmentedControlProps['items'],
+) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const [selectedLeft, setSelectedLeft] = useState(0);
+  const [selectedWidth, setSelectedWidth] = useState(0);
+
+  useEffect(() => {
+    const updateSelectedStates = () => {
+      if (ref.current) {
+        const selectedControl = ref.current.children[selectedIndex] as HTMLElement | undefined;
+        if (!selectedControl) {
+          return;
+        }
+        setSelectedLeft(selectedControl.offsetLeft);
+        setSelectedWidth(selectedControl.offsetWidth);
+      }
+    };
+
+    updateSelectedStates();
+    window.addEventListener('resize', updateSelectedStates);
+    return () => {
+      window.removeEventListener('resize', updateSelectedStates);
+    };
```

Comment:
We could consider using a `ResizeObserver` on the segmented control, so we don't have to listen for the whole page (since a page resize may occur without affecting the width of the segmented control)